src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Status.tsx

```diff
@@ -54,16 +59,38 @@ export const Status = () => {
   const confirmedTransaction = useAppSelector(gs => selectTxById(gs, serializedTxIndex))
 
   useEffect(() => {
-    if (confirmedTransaction && confirmedTransaction.status !== 'Pending' && dispatch) {
-      dispatch({
-        type: ThorchainSaversDepositActionType.SET_DEPOSIT,
-        payload: {
-          txStatus: confirmedTransaction.status === 'Confirmed' ? 'success' : 'failed',
-          usedGasFee: confirmedTransaction.fee?.value,
-        },
-      })
+    if (!accountId) return
+
+    if (confirmedTransaction && confirmedTransaction.status !== 'Pending' && contextDispatch) {
+      ;(async () => {
+        // Artificial longer completion time, since THORChain Txs take around 15s after confirmation to be picked in the API
+        // This way, we ensure "View Position" actually routes to the updated position
+        await new Promise(resolve => setTimeout(resolve, 17000))
```

Comment:
@reallybeard as polish we may want to add a bar/countdown or something, we can't avoid tx confirmation times, and 17s with no feedback is really long for users.